examples/terraform-aws-s3-example/variables.tf

```diff
@@ -17,6 +17,11 @@
 # These parameters have reasonable defaults.
 # ---------------------------------------------------------------------------------------------------------------------
 
+variable "with_policy" {
+  description = "If set to `true`, the bucket will be created with a bucket policy."
+  default = "false"
```

Comment:
Can you switch to using bools? (`false` without quotes).

Reply:
Terraform best practices suggest this as an anti-pattern:

https://www.terraform.io/docs/configuration-0-11/variables.html#booleans

I will make the change, but just trying to stick with best practices.  Thoughts?

Reply:
Terraform 12 has a native bool type: https://www.terraform.io/docs/configuration/types.html#primitive-types. We're getting ready to upgrade to tf12, so just want to add these compatibilities in new code.

Also, the specific anti pattern is relying on the conversion rules of Terraform (e.g `true` => `1` and `false` => `0`). So relying on the bool in a conditional is actually a valid (and probably the only) case where bool makes sense.

Reply:
Sounds good.  Implemented.